resolwe_bio/processes/reads_processing/trimgalore.py

```diff
@@ -0,0 +1,442 @@
+
+        params = [
+            "--paired",
+            "--cores",
+            2,  # Actual core usage is 9
+            inputs.quality_trim.phred,
+            "--length",
+            inputs.quality_trim.min_length,
+            "--stringency",
+            inputs.adapter_trim.stringency,
+            "-e",
+            inputs.adapter_trim.error_rate,
+        ]
+
+        if inputs.quality_trim.nextseq:
+            params.extend(["--nextseq", inputs.quality_trim.nextseq])
+        else:
+            params.extend(["--quality", inputs.quality_trim.quality])
+
+        if inputs.quality_trim.max_n:
+            params.extend(["--max_n", inputs.quality_trim.max_n])
+        if inputs.quality_trim.retain_unpaired:
+            params.extend(
+                [
+                    "--retain_unpaired",
+                    "--length_1",
+                    inputs.quality_trim.unpaired_len_1,
+                    "--length_2",
+                    inputs.quality_trim.unpaired_len_2,
+                ]
+            )
+        if inputs.quality_trim.clip_r1:
+            params.extend(["--clip_R1", inputs.quality_trim.clip_r1])
+        if inputs.quality_trim.clip_r2:
+            params.extend(["--clip_R2", inputs.quality_trim.clip_r2])
+        if inputs.quality_trim.three_prime_r1:
+            params.extend(["--three_prime_clip_R1", inputs.quality_trim.three_prime_r1])
+        if inputs.quality_trim.three_prime_r2:
+            params.extend(["--three_prime_clip_R2", inputs.quality_trim.three_prime_r2])
+
+        if inputs.adapter_trim.adapter:
+            for adapter in inputs.adapter_trim.adapter:
+                params.extend(["--adapter", adapter])
+        if inputs.adapter_trim.adapter_2:
+            for adapter in inputs.adapter_trim.adapter_2:
+                params.extend(["--adapter2", adapter])
+        if inputs.adapter_trim.adapter_file_1:
+            params.extend(
+                ["--adapter", f"file:{inputs.adapter_trim.adapter_file_1.fasta.path}"]
+            )
+        if inputs.adapter_trim.adapter_file_2:
+            params.extend(
+                ["--adapter2", f"file:{inputs.adapter_trim.adapter_file_2.fasta.path}"]
+            )
+        if inputs.adapter_trim.universal_adapter:
+            params.append(inputs.adapter_trim.universal_adapter)
+            if (
+                inputs.adapter_trim.adapter
+                or inputs.adapter_trim.adapter_2
+                or inputs.adapter_trim.adapter_file_1
+                or inputs.adapter_trim.adapter_file_2
+            ):
+                self.error(
+                    "You can not supply custom adapter sequence and use "
+                    "the universal adapter sequence."
+                )
+
+        if inputs.hard_trim.trim_5 or inputs.hard_trim.trim_3:
+            self.info(
+                "Only hard trimming was performed. Skipped quality and adapter trimming"
+            )
+        if inputs.hard_trim.trim_5:
```

Comment:
Maybe nest this two `if`s under the above `if inputs.hard_trim.trim_5 or inputs.hard_trim.trim_3:` since they belong together.